rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java

```diff
@@ -637,29 +637,29 @@ public void acquireAllNodes(Map<String, ?> nodeConfiguration) {
      * @param nodeUrl the url of the node to be released
      */
     public BooleanWrapper removeNode(String nodeUrl, Client initiator) {
-
+        Node node = null;
         //verifying if node is already in the list,
         //node could have fallen between remove request and the confirm
         if (this.nodes.containsKey(nodeUrl)) {
-            logger.info("[" + name + "] removing node : " + nodeUrl);
-            Node node = nodes.remove(nodeUrl);
+            logger.info("[" + this.name + "] removing node: " + nodeUrl);
+            node = this.nodes.remove(nodeUrl);
+        } else if (this.downNodes.containsKey(nodeUrl)) {
+            logger.info("[" + this.name + "] removing down node: " + nodeUrl);
+            node = this.downNodes.remove(nodeUrl);
+        }
+
+        if (node == null) {
+            logger.error("[" + this.name + "] cannot remove node: " + nodeUrl + " because it is unknown");
+            return new BooleanWrapper(false);
+        } else {
```

Comment:
small side effect is that a warning message will be printed by TopologyManager when removing a down node. Not a big deal.